api/v1alpha1/meshgateway.proto

```diff
@@ -0,0 +1,119 @@
+// Copyright 2021 Cisco Systems, Inc. and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "google/protobuf/wrappers.proto";
+import "api/v1alpha1/common.proto";
+import "gogoproto/gogo.proto";
+import "google/api/field_behavior.proto";
+import "k8s.io/api/core/v1/generated.proto";
+
+// $schema: istio-operator.api.v1alpha1.MeshGatewaySpec
+// $title: Istio Mesh Gateway Spec
+// $description: Istio Mesh Gateway descriptor
+
+package istio_operator.v2.api.v1alpha1;
+
+option go_package = "github.com/banzaicloud/istio-operator/v2/api/v1alpha1";
+
+// MeshGateway defines an Istio ingress or egress gateway
+//
+// <!-- crd generation tags
+// +cue-gen:MeshGateway:groupName:servicemesh.cisco.com
+// +cue-gen:MeshGateway:version:v1alpha1
+// +cue-gen:MeshGateway:storageVersion
+// +cue-gen:MeshGateway:annotations:helm.sh/resource-policy=keep
+// +cue-gen:MeshGateway:subresource:status
+// +cue-gen:MeshGateway:scope:Namespaced
+// +cue-gen:MeshGateway:resource:shortNames=mgw,meshgw
+// +cue-gen:MeshGateway:printerColumn:name=Age,type=date,JSONPath=.metadata.creationTimestamp,description="CreationTimestamp is a timestamp
+// representing the server time when this object was created. It is not guaranteed to be set in happens-before order across separate operations.
+// Clients may not set this value. It is represented in RFC3339 form and is in UTC.
+// Populated by the system. Read-only. Null for lists. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata"
+// +cue-gen:MeshGateway:preserveUnknownFields:false
+// -->
+//
+// <!-- go code generation tags
+// +genclient
+// +k8s:deepcopy-gen=true
+// -->
+message MeshGatewaySpec {
+    // Contains the intended version for the Mesh Gateway.
+    // +kubebuilder:validation:Pattern=^1.
+    string version = 1 [(gogoproto.jsontag) = "version", (google.api.field_behavior) = REQUIRED];
```

Comment:
Why should we have a `version` for the MGW resource, what would be the purpose of that? A gateway could be bound to any Istio control plane, which can be set below in the config. During an upgrade e.g. to user should change the version too to the upgraded version? Or should it be left intact? In that case it might be confusing that the version is different than the corresponding istio control plane version.

Reply:
It's a copy-paste error, thanks for noticing :)